src/thunderbots/software/test/ai/passing/pass_generator.cpp

```diff
@@ -7,20 +7,66 @@
 #include <gtest/gtest.h>
 #include <string.h>
 
+#include "ai/passing/evaluation.h"
 #include "test/test_util/test_util.h"
 
 using namespace AI::Passing;
 using namespace std::chrono_literals;
 
 class PassGeneratorTest : public testing::Test
 {
-   protected:
    protected:
     virtual void SetUp()
     {
         world = ::Test::TestUtil::createBlankTestingWorld();
         world.updateFieldGeometry(::Test::TestUtil::createSSLDivBField());
-        pass_generator = std::make_shared<PassGenerator>(0.05, world, Point(1, 0));
+        pass_generator = std::make_shared<PassGenerator>(world, Point(0, 0));
+    }
+
+    /**
+     * Wait for the pass generator to converge to a pass
+     *
+     * If the pass generator does not converge within `max_num_seconds`, this will
+     * cause the test to fail
+     *
+     * @param pass_generator Modified in-place
+     * @param max_score_diff The maximum absolute difference between the scores of three
+     *                       consecutive optimized passes before this function returns
+     * @param max_num_seconds The maximum number of seconds that the pass optimizer
+     *                        can run for before this function returns
+     */
+    static void waitForConvergence(std::shared_ptr<PassGenerator> pass_generator,
+                                   double max_score_diff, int max_num_seconds)
+    {
+        int seconds_so_far     = 0;
+        double curr_score      = 0;
+        double prev_score      = 0;
+        double prev_prev_score = 0;
+        do
+        {
+            prev_prev_score = prev_score;
+            prev_score      = curr_score;
+
+            std::this_thread::sleep_for(1s);
+            seconds_so_far++;
+
+            auto curr_pass_and_score = pass_generator->getBestPassSoFar();
+            curr_score               = curr_pass_and_score.second;
+            std::cout << curr_score << std::endl;
```

Comment:
Probably don't want to leave printouts in the test

Reply:
Leaving this one here so i can see what's going on when CI inevitably causes issues for these tests, but yup, please leave this unresolved so I remember to remove them before merging.